Windows/debugger.cpp

```diff
@@ -1536,8 +1536,35 @@ DebuggerStatus Debugger::Attach(unsigned int pid, uint32_t timeout) {
   attach_mode = true;
 
   if (!DebugActiveProcess(pid)) {
-    FATAL("Could not attach to the process.\n"
-          "Make sure the process exists and you have permissions to debug it.\n");
+    DWORD error_code = GetLastError();
+    SAY("DebugActiveProcess(%d) failed with error code = %d\n", error_code);
+
+    if(error_code == 5) {
+      HANDLE hToken = NULL;
+      LUID luid;
+      TOKEN_PRIVILEGES tp;
+      SAY("Try again after enabling 'SeDebugPrivilege'\n");
+
+      if(OpenProcessToken(GetCurrentProcess(), TOKEN_ADJUST_PRIVILEGES | TOKEN_QUERY, &hToken)) {

```

Comment:
prefer the style of

```
if(!Something()) {
  FATAL();
}
if(!SomethingElse()) {
  FATAL();
}
```

over

```
if(Something()) {
  if(Something()) {
    ...
  } else {
    FATAL();
  }
} else {
  FATAL();
}
```

As the first one is more readable. The second creates too many nested ifs and it's difficult to tell which else refers to which if.